include/xtensor/xexception.hpp

```diff
@@ -151,14 +151,39 @@ namespace xt
         }
     }
 
+    /*******************
+     * check_dimension *
+     *******************/
+
+    template <class S, class... Args>
+    inline void check_dimension(const S& shape, Args... args)
+    {
+        if (sizeof...(Args) > shape.size())
+        {
+            throw std::out_of_range("Number of arguments (" + std::to_string(sizeof...(Args)) + ") us greater "
+                + "than the number of dimensions (" + std::to_string(shape.size()) + ")");
+        }
+    }
```

Comment:
Should check_access and `check_dimension` be in `xstrides`?

Reply:
I am not sure we want to check the dimension at all, since we have a clear rule about what to do when too many arguments are provided.

Reply:
Sorry I missed the comments before merging